restore/data.go

```diff
@@ -301,7 +304,7 @@ func restoreDataFromTimestamp(fpInfo filepath.FilePathInfo, dataEntries []toc.Co
 					}
 				}
 				if err == nil {
-					err = restoreSingleTableData(ctx, &fpInfo, entry, tableName, whichConn)
+					err = restoreSingleTableData(ctx, cancel, &fpInfo, entry, tableName, whichConn)
```

Comment:
Passing context cancel function isn't recommended

Reply:
Why? https://github.com/arenadata/gpbackup/blob/7087e491a1a01f0b71c49c31f48abd382f83dc10/restore/data.go#L264

Reply:
PS I don't review patch made this change. So this part of question should be addressed to the author and reviewers.

I relied in my comment on the article from [here](https://p.agnihotry.com/post/understanding_the_context_package_in_golang/). As far I understand, passing cancel function as an argument violates responsibility principle, makes code more tightly couple and error prone.

Reply:
> passing cancel function as an argument violates responsibility principle, makes code more tightly couple and error prone.

Then this is another argument against the current error file check every 5 seconds!

Reply:
But it's a problem for the separate task. Discussed one was fixed.